server/plugin.go

```diff
@@ -115,7 +115,10 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	p.initializeAPI()
-	p.initializeTelemetry()
+
+	if p.API.GetConfig().LogSettings.EnableDiagnostics != nil {
```

Comment:
As a side note, we would want to also ensure it is set to `true`

```suggestion
	if p.API.GetConfig().LogSettings.EnableDiagnostics != nil && *p.API.GetConfig().LogSettings.EnableDiagnostics {
```